openmdao/core/system.py

```diff
@@ -808,31 +812,34 @@ def _apply_linear_jac(self, params, unknowns, dparams, dunknowns, dresids, mode)
                              if p not in dparams])
 
         for (unknown, param), J in iteritems(self._jacobian_cache):
+
             if param in states:
                 arg_vec = dunknowns
             else:
                 arg_vec = dparams
 
             # Vectors are flipped during adjoint
-
             try:
                 if isvw:
                     if fwd:
                         vec = dresids._flat(unknown)
-                        vec += J.dot(arg_vec._flat(param))
+                        vec += J.dot(arg_vec[param].flat)
```

Comment:
I'm curious about this change.  By calling arg_vec[param].flat, you're first retrieving the unflattened value then flattening it, where before you were retrieving the flat value directly.